SU2_CFD/include/solvers/CSolver.hpp

```diff
@@ -0,0 +1,4995 @@
+   */
+  void CompleteComms(CGeometry *geometry,
+                     CConfig *config,
+                     unsigned short commType);
+
+  /*!
+   * \brief Routine to load a solver quantity into the data structures for MPI periodic communication and to launch non-blocking sends and recvs.
+   * \param[in] geometry - Geometrical definition of the problem.
+   * \param[in] config   - Definition of the particular problem.
+   * \param[in] val_periodic_index - Index for the periodic marker to be treated (first in a pair).
+   * \param[in] commType - Enumerated type for the quantity to be communicated.
+   */
+  void InitiatePeriodicComms(CGeometry *geometry,
+                             CConfig *config,
+                             unsigned short val_periodic_index,
+                             unsigned short commType);
+
+  /*!
+   * \brief Routine to complete the set of non-blocking periodic communications launched by InitiatePeriodicComms() and unpacking of the data in the solver class.
+   * \param[in] geometry - Geometrical definition of the problem.
+   * \param[in] config   - Definition of the particular problem.
+   * \param[in] val_periodic_index - Index for the periodic marker to be treated (first in a pair).
+   * \param[in] commType - Enumerated type for the quantity to be unpacked.
+   */
+  void CompletePeriodicComms(CGeometry *geometry,
+                             CConfig *config,
+                             unsigned short val_periodic_index,
+                             unsigned short commType);
+
+  /*!
+   * \brief Set number of linear solver iterations.
+   * \param[in] val_iterlinsolver - Number of linear iterations.
+   */
+  inline void SetIterLinSolver(unsigned short val_iterlinsolver) { IterLinSolver = val_iterlinsolver; }
+
+  /*!
+   * \brief Set the final linear solver residual.
+   * \param[in] val_reslinsolver - Value of final linear solver residual.
+   */
+  inline void SetResLinSolver(su2double val_reslinsolver) { ResLinSolver = val_reslinsolver; }
+
+  /*!
+   * \brief Set the value of the max residual and RMS residual.
+   * \param[in] val_iterlinsolver - Number of linear iterations.
+   */
+  void SetResidual_RMS(CGeometry *geometry, CConfig *config);
+
+  /*!
+   * \brief Communicate the value of the max residual and RMS residual.
+   * \param[in] val_iterlinsolver - Number of linear iterations.
+   */
+  void SetResidual_BGS(CGeometry *geometry, CConfig *config);
+
+  /*!
+   * \brief Set the value of the max residual and RMS residual.
+   * \param[in] val_iterlinsolver - Number of linear iterations.
+   */
+  virtual void ComputeResidual_Multizone(CGeometry *geometry, CConfig *config);
+
+  /*!
+   * \brief Move the mesh in time
+   */
+  inline virtual void SetDualTime_Mesh(void){ }
+
+  /*!
+   * \brief Store the BGS solution in the previous subiteration in the corresponding vector.
+   */
+  void UpdateSolution_BGS(CGeometry *geometry, CConfig *config);
+
```

Comment:
Is there a reason why this is not declared virtual? There are instantiations of it in `CDiscAdjFEASolver` and in CDiscAdjSolver`

Reply:
Hmmm curious... I 🤦 myself for a minute thinking this was the explanation to something important, turns out it is not very relevant.
Please delete the versions of the method in those two DiscAdj solvers, they only affect the legacy adjoint FSI driver (namely the computation of multizone residuals) and that is on its way out so...
You may be able to catch more of these with warning `-Woverloaded-virtual`